### PR TITLE
dev/core#117 Replace deprecated each with foreach in CRM/Utils/Token.php

### PR DESCRIPTION
**Pull request by seamuslee001**
Overview
----------------------------------------
Another PHP7.2 compatibility PR, title says it all

ping @eileenmcnaughton @monishdeb

### CODE REVIEW

File: CRM/Utils/Token.php [hunk 1]
```diff
@@ -1404,7 +1404,7 @@ public static function replaceGreetingTokens(&$tokenString, $contactDetails = NU
         $greetingTokens = $remainingTokens;
         reset($greetingTokens);
         $greetingsReturnProperties = array();
-        while (list($key) = each($greetingTokens)) {
+        foreach ($greetingTokens as $key => $value) {
           $props = array_flip(CRM_Utils_Array::value($key, $greetingTokens));
```
Comment (monishdeb) [inline CRM/Utils/Token.php]: Or 
```patch
- foreach ($greetingTokens as $key => $value) {
-    $props = array_flip(CRM_Utils_Array::value($key, $greetingTokens));
+ foreach ($greetingTokens as $value) {
+   $props = array_flip($value);
```
??
As $key is only used in 1 place and $value is no longer used

Comment (seamuslee001) [reply CRM/Utils/Token.php]: hmm that could work i guess i was just aiming at a minimalist replacement process but that could work

**Comment by seamuslee001** [comment]
@monishdeb updated

**Comment by monishdeb** [comment]
Jenkins test this please

**Comment by monishdeb** [comment]
Tested, working fine. Played with greeting token, doesn't break the token replacement. Hence merging

**Pull request merged**